### PR TITLE
Garden Cat (Account Shared Inventory)

### PR DESCRIPTION
**Pull request by stratic-dev**
- Implemented two packets to fix the garden cat. (Shared Account Box)
- Added readme for migrations

### CODE REVIEW

File: Erupe/server/channelserver/handlers.go [hunk 1]
```diff
@@ -356,9 +357,100 @@ func handleMsgMhfAcquireTitle(s *Session, p mhfpacket.MHFPacket) {}
...
+			for i := 0; i < amount; i++ {
+				bf.WriteUint32(binary.BigEndian.Uint32(boxContents[i*4 : i*4+4]))
+				if i+1 != amount {
+					bf.WriteUint64(0x00)
+				}
+			}
+		}
+	}
+	doAckBufSucceed(s, pkt.AckHandle, bf.Data())
+	doAckBufSucceed(s, pkt.AckHandle, bf.Data())
```
Comment (sekaiwish) [inline Erupe/server/channelserver/handlers.go:389]: Here you are sending the buffered data twice. I don't think this causes any issues but it should be cut.

File: Erupe/network/mhfpacket/msg_mhf_update_union_item.go [hunk 2]
```diff
@@ -1,15 +1,27 @@
...
+	"errors"
 
- 	"github.com/Solenataris/Erupe/network/clientctx"
-	"github.com/Solenataris/Erupe/network"
 	"github.com/Andoryuuta/byteframe"
+	"github.com/Solenataris/Erupe/network"
+	"github.com/Solenataris/Erupe/network/clientctx"
 )
 
+type Item struct {
```
Comment (sekaiwish) [inline Erupe/network/mhfpacket/msg_mhf_update_union_item.go:11]: The Item struct being redeclared causes a compilation error.

File: Erupe/server/channelserver/handlers.go [hunk 3]
```diff
@@ -356,9 +357,100 @@ func handleMsgMhfAcquireTitle(s *Session, p mhfpacket.MHFPacket) {}
 
 func handleMsgMhfEnumerateTitle(s *Session, p mhfpacket.MHFPacket) {}
 
-func handleMsgMhfEnumerateUnionItem(s *Session, p mhfpacket.MHFPacket) {}
+type Item struct {
```
Comment (sekaiwish) [inline Erupe/server/channelserver/handlers.go:360]: Third declaration of Item struct.

**Review by sekaiwish — CHANGES_REQUESTED** [review]

File: Erupe/server/channelserver/handlers.go (hunk 3 above)
Comment (stratic-dev) [reply Erupe/server/channelserver/handlers.go:360]: Background conversation has been had, There was confusion between this fork and his own.  I'm going to combine both struts and move them to a common place. Wish has suggested handlers.go. for now

File: Erupe/server/channelserver/handlers.go (hunk 1 above)
Comment (stratic-dev) [reply Erupe/server/channelserver/handlers.go:389]: Good catch ill remove.

**Pull request merged**